library/network/src/modules/NetworkInterfaces.rb

```diff
@@ -789,7 +789,7 @@ def adapt_old_config!
     # @param [Hash] a map with netconfig (ifcfg) configuration
     #
     def add_device(device, ifcfg)
-      devtype = GetTypeFromIfcfg(ifcfg) || GetType(device)
+      devtype = GetType(device)
```

Comment:
Difference is that ```GetTypeFromIfcfgOrName``` uses ifcfg file or name for device type detection. It is just a subset of whole type detection process which is implemented in ```GetType```. ```GetType``` is the only function which should be present in public API (but it is not bcs of ycp). Particular example of collision is described in the commit message.